modules/migrations/github.go

```diff
@@ -116,8 +132,18 @@ func (g *GithubDownloaderV3) SetContext(ctx context.Context) {
 }
 
 func (g *GithubDownloaderV3) sleep() {
-	for g.rate != nil && g.rate.Remaining <= GithubLimitRateRemaining {
-		timer := time.NewTimer(time.Until(g.rate.Reset.Time))
+	var recentIdx int
+	var maxRemaining int
+	for i := 0; i < len(g.clients); i++ {
+		if g.rates[i] != nil && g.rates[i].Remaining > maxRemaining {
+			maxRemaining = g.rates[i].Remaining
+			recentIdx = i
+		}
+	}
+	g.curClientIdx = recentIdx
```

Comment:
The function name `sleep` is not accurate now. It also changes internal state, eg: `curClientIdx`

Maybe this function should have a new name.